scrapy/crawler.py

```diff
@@ -105,24 +104,8 @@ def __init__(
             crawler=self,
         )
```

Comment:
We might want to move the fingerprinter as well, specially if we move addons, to allow addons to change the fingerprinter.

I also wonder if there is more stuff we could move. Like logging the overriden settings, and *maybe* stats (not sure what that could break).